platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/hash/VirtualHasher.java

```diff
@@ -400,10 +409,21 @@ public Hash hash(
                     while (curStackPath < Math.min(curPath, lastPathInCurStackChunk)) {
                         final ChunkHashTask t = map.remove(curStackPath);
                         assert t != null;
-                        t.setData(null);
+                        t.complete();
                         curStackPath++;
                     }
-                    stack[curRank] = INVALID_PATH;
+                    /*
+                       It may happen that curPath is actually in the same chunk as lastPathInCurStackChunk.
```

Comment:
"... in the same chunk as stack[curRank]"

Reply:
Done